datafusion/functions-aggregate/src/count.rs

```diff
@@ -138,6 +138,10 @@ impl AggregateUDFImpl for Count {
             return Ok(Box::new(CountAccumulator::new()));
         }
 
+        if acc_args.input_exprs.len() > 1 {
```

Comment:
I'm thinking should it parser issue? Is there any SQL conventions supporting the multiple expressions for COUNT(DISTINCT)?

Reply:
just checked https://chicagobusinesspress.com/download_ancillary?book=22&ancillary=157 for SQL 2016 standard, I can only 1 col supported in syntax. I think we can go with a patch but followup has to be in sqlparser-rs

Reply:
As far as I know, [Spark](https://spark.apache.org/docs/latest/api/sql/index.html#count) and [MySQL](https://dev.mysql.com/doc/refman/8.4/en/aggregate-functions.html#function_count-distinct) support this syntax.

Reply:
If any of the dialects allow the syntax I think the sqlparser-rs AST will need to support multiple args.

This is kind of explained https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#syntax-vs-semantics (where what is a syntax error in some systems and a semantic one in others needs to be enforced as a semantic error)

Thus I think the right thing to do here is check and error in DataFusion, as @jonahgao  's PR does

Reply:
That is correct, Spark supports it, I double checked it. Maybe it is SQL 2023 standard, but if any of dialects supports this syntax then parser is not expected to restrict it and this PR is correct

Reply:
I would say "SQL Standard" in quotes -- because even though there is such a thing I don't think any database follows it exactly. It is more like "SQL Standard Suggestions-ish" or something 🤣